cpp/src/arrow/json/from_string.cc

```diff
@@ -1030,13 +1029,13 @@ Status DictArrayFromJSONString(const std::shared_ptr<DataType>& type,
                         ArrayFromJSONString(dictionary_type.index_type(), indices_json));
   ARROW_ASSIGN_OR_RAISE(auto dictionary, ArrayFromJSONString(dictionary_type.value_type(),
                                                              dictionary_json));
-
-  return DictionaryArray::FromArrays(type, std::move(indices), std::move(dictionary))
-      .Value(out);
+  ARROW_ASSIGN_OR_RAISE(auto out, DictionaryArray::FromArrays(type, std::move(indices),
+                                                              std::move(dictionary)));
+  return out;
```

Comment:
Cam we simplify this?

```suggestion
  return DictionaryArray::FromArrays(type, std::move(indices), std::move(dictionary)));
```